lib/utils.js

```diff
@@ -0,0 +1,93 @@
+/** @babel */
+
+import stdPath from 'path';
+
+
+/**
+ * Generates the return value for the wrapper property on first access
+ * and caches it on the object. All future calls return the cached value
+ * instead of re-calculating it.
+ */
+export function cachedProperty(target, key, descriptor) {
+    let getter = descriptor.get;
+    let cached_key = `_${key}_cached`;
```

Comment:
I'm not sure if they work in your environment, but ES6 Symbols are a good fit for what you're doing here. They would have 0 chance of colliding with anything.